js/ui/layout.js

```diff
@@ -84,23 +84,12 @@ var LayoutManager = new Lang.Class({
         global.stage.no_clear_hint = true;
 
         // Set up stage hierarchy to group all UI actors under one container.
-        this.uiGroup = new Shell.GenericContainer({ name: 'uiGroup' });
-        this.uiGroup.connect('allocate',
-                        function (actor, box, flags) {
-                            let children = actor.get_children();
-                            for (let i = 0; i < children.length; i++)
-                                children[i].allocate_preferred_size(flags);
-                        });
-        this.uiGroup.connect('get-preferred-width',
-                        function(actor, forHeight, alloc) {
-                            let width = global.stage.width;
-                            [alloc.min_size, alloc.natural_size] = [width, width];
-                        });
-        this.uiGroup.connect('get-preferred-height',
-                        function(actor, forWidth, alloc) {
-                            let height = global.stage.height;
-                            [alloc.min_size, alloc.natural_size] = [height, height];
-                        });
+        this.uiGroup = new St.Widget({ name: 'uiGroup' });
+        this.uiGroup.set_flags(Clutter.ActorFlags.NO_LAYOUT);
```

Comment:
Does this not belong to a separate commit?

Reply:
Ouch. This was a performance touch from a commit upstream, and it sneaked there while i was rebasing the patchset.